media-tv/plex-media-server/plex-media-server-1.13.9.5456.ebuild

```diff
@@ -0,0 +1,135 @@
+
+QA_DESKTOP_FILE="usr/share/applications/plexmediamanager.desktop"
+QA_PREBUILT="*"
+QA_MULTILIB_PATHS=(
+	"usr/lib/${_APPNAME}/.*"
+	"usr/lib/${_APPNAME}/Resources/Python/lib/python2.7/.*"
+)
+
+S="${WORKDIR}"
+
+pkg_setup() {
+	enewgroup ${_USERNAME}
+	enewuser ${_USERNAME} -1 /bin/bash /var/lib/${_APPNAME} "${_USERNAME},video"
+
+	python-single-r1_pkg_setup
+}
+
+src_unpack() {
+	unpack_deb ${A}
+}
+
+src_prepare() {
+	eapply "${FILESDIR}"/plexmediamanager.desktop.patch
+	default
+}
+
+src_install() {
+	# Move the config to the correct place
+	local CONFIG_VANILLA="/etc/default/"${_APPNAME}""
+	local CONFIG_PATH="/etc/${_SHORTNAME}"
+	dodir "${CONFIG_PATH}"
+	insinto "${CONFIG_PATH}"
+	doins "${CONFIG_VANILLA#/}"
+	sed -e "s#${CONFIG_VANILLA}#${CONFIG_PATH}/$(basename "${CONFIG_VANILLA}")#g" \
+		-i "${S}"/usr/sbin/start_pms || die
+
+	# Remove Debian specific files
+	rm -rf "usr/share/doc" || die
+
+	# Copy main files over to image and preserve permissions so it is portable
+	cp -rp usr/ "${ED}" || die
+
+	# Make sure the logging directory is created
+	local LOGGING_DIR="/var/log/pms"
+	dodir "${LOGGING_DIR}"
+	chown "${_USERNAME}":"${_USERNAME}" "${ED%/}/${LOGGING_DIR}" || die
+
+	# Create default library folder with correct permissions
+	local DEFAULT_LIBRARY_DIR="/var/lib/${_APPNAME}"
+	dodir "${DEFAULT_LIBRARY_DIR}"
+	chown "${_USERNAME}":"${_USERNAME}" "${ED%/}/${DEFAULT_LIBRARY_DIR}"
+
+	# Install the OpenRC init/conf files depending on avahi.
+	if use avahi; then
+		doinitd "${FILESDIR}/init.d/${PN}"
+	else
+		cp "${FILESDIR}/init.d/${PN}" "${S}/${PN}" || die;
+		sed -e '/depend/ s/^#*/#/' -i "${S}/${PN}"
+		sed -e '/need/ s/^#*/#/' -i "${S}/${PN}"
+		sed -e '1,/^}/s/^}/#}/' -i "${S}/${PN}"
+		doinitd "${S}/${PN}"
+	fi
+
+	_remove_execstack_markings
+
+	doconfd "${FILESDIR}/conf.d/${PN}"
+
+	# Install systemd service file
+	local INIT_NAME="${PN}.service"
+	local INIT="${FILESDIR}/systemd/${INIT_NAME}"
+	systemd_newunit "${INIT}" "${INIT_NAME}"
+
+	_handle_multilib
+}
+
+pkg_postinst() {
+	einfo ""
+	elog "Plex Media Server is now installed. Please check the configuration file in /etc/${_SHORTNAME}/${_APPNAME} to verify the default settings."
+	elog "To start the Plex Server, run 'rc-config start plex-media-server', you will then be able to access your library at http://<ip>:32400/web/"
+}
+
+# Remove execstack flags from some libraries/executables
+# so that it works in hardened setups.
+_remove_execstack_markings() {
+	for f in "${EXECSTACKED_BINS[@]}"; do
+	# Unquoting 'f' so that expansion works.
+		fix-gnustack -f ${f} > /dev/null
+	done
+}
+
+# Finds out where the library directory is for this system
+# and handles ldflags as to not break library dependencies
+# during rebuilds.
+_handle_multilib() {
+	# Prevent revdep-rebuild, @preserved-rebuild breakage
+	cat > "${T}"/66plex <<-EOF || die
+		LDPATH="${EPREFIX}/usr/$(get_libdir)/plexmediaserver"
+	EOF
+
+	doenvd "${T}"/66plex
```

Comment:
I **think** this whole phase can be accomplished with one `newenvd` line, instead of cat - EOF - doenvd.

Reply:
I dont know if this would work on multilib boxes, then. Would it?